src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractAISkeleton.java

```diff
@@ -51,12 +51,18 @@
     protected AbstractAISkeleton(@NotNull final J job)
     {
         super();
+
+        if (!job.getCitizen().getCitizenEntity().isPresent())
+        {
+            throw new IllegalArgumentException("Cannot instantiate a AI from a Job that is attached to a Citizen without entity.");
+        }
+
         this.targetList = new ArrayList<>();
         setMutexBits(MUTEX_MASK);
         this.job = job;
-        this.worker = this.job.getCitizen().getCitizenEntity();
+        this.worker = this.job.getCitizen().getCitizenEntity().get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)